### ISSUE TITLE
Mission progress not triggered when mission is uploaded from QGC

### ISSUE THREAD

**Issue by JeneferSuresh** [mavlink/MAVSDK#1804]
Hi
I try to run the QGC on the laptop 
PX4 by gazebo

When I try to use Mission::subscribe_mission_progress, my callback does not get invoked.

I tried to use the fly_mission.cpp example, even in that, the callback does not get invoked.

Any reason, or any pre-requsites?

**Comment by JonasVautherin** [comment]
> Or just mavsdk should be enough to handle that?

Not sure I understand that. MAVSDK needs to know about the mission, either by uploading it to the drone, or by downloading it from the drone (e.g. when QGC was the one uploading). Once MAVSDK knows about the mission, the progress subscription will work.

So from your ROS node you need to do what the example does: download (or upload) _and_ subscribe to the mission progress.

Does that answer your question?

**Comment by JeneferSuresh** [comment]
I do the same. Download from drone and subscribe to track progress.

since subscribing listen to certain mavlink messages. I was confused if I have to include those mavlink headers.

Right now, I include mavsdk in package.xml and cmake.

**Comment by JonasVautherin** [comment]
Does it mean that it's working? Can we close this issue?

**Comment by JeneferSuresh** [comment]
No, am sorry. Still we are not able to track the progress. But I can see that the mission gets downloaded correctly.
Am not sure what am missing

**Comment by JonasVautherin** [comment]
Can you then explain again what your problem is?

> I tried to download and track the progress in the example, then it works.

Seems to me that it is working when using MAVSDK, but you have issues with your ROS code. Is that correct?

**Comment by JeneferSuresh** [comment]
Sorry for the late response.
I was not able to link up mission raw plugin with my ros node, the initial messages get listened. Then once the node gets started up, it does not listen anymore.
I am not sure of the reason.

i then managed to create another plugin to listen only for missed_reached message. I was able to listen and track the mission even without downloading the mission.

Now i try to subscribe to the updates. i get into a segmentation fault when trying to invoke callback  method at system_impl.cpp.


Does it have to do anything with passing back method as reference variable. Or anything i have to do differently when subscribing  from ros2 node? I tried passing lambda, method, same result.
